lib/toccatore/usage_update.rb

```diff
@@ -0,0 +1,171 @@
+require_relative 'base'
+
+
+module Toccatore
+  class UsageUpdate < Base
+    include Toccatore::Queue
+    LICENSE = "https://creativecommons.org/publicdomain/zero/1.0/"
+
+
+    def initialize options={}
+      @sqs = queue options
+    end
+
+    def queue_jobs(options={})
+
+      total = get_total(options)
+      
+      while total > 0
+        # walk through paginated results
+        total_pages = (total.to_f / job_batch_size).ceil
+        error_total = 0
+
+        (0...total_pages).each do |page|
+          options[:offset] = page * job_batch_size
+          options[:total] = total
+          error_total += process_data(options)
+        end
+        text = "#{total} works processed with #{error_total} errors for Usage Reports Queue"
+      end
+
+      #   text = "No works found for in the queue."
+      # end
+
+      # send slack notification
+      options[:level] = total > 0 ? "good" : "warning"
+      options[:title] = "Report for #{source_id}"
+      send_notification_to_slack(text, options) if options[:slack_webhook_url].present?
+
+      # return number of works queued
+      total
+    end
+
+    def process_data(options = {})
+      message = get_message
+      data = get_data(message)
+      data = parse_data(data, options)
+
+      return [OpenStruct.new(body: { "data" => [] })] if data.empty?
+
+      push_data(data, options)
+      delete_message message
+    end
+
+    def get_data reponse 
+      return OpenStruct.new(body: { "errors" => "Queue is empty" }) if reponse.messages.empty?
+
+      body = JSON.parse(reponse.messages[0].body)
+      Maremma.get(body["report_id"])
+    end
+
+
+    # method returns number of errors
+    def push_data(items, options={})
+      if items.empty?
+        puts "No works found in the Queue."
+        0
+      elsif options[:access_token].blank?
+        puts "An error occured: Access token missing."
+        options[:total]
+      else
+        error_total = 0
+        Array(items).each do |item|
+          error_total += push_item(item, options)
+        end
+        error_total
+      end
+    end
+
+    def metrics_url
+      "https://metrics.test.datacite.org/reports/"
```

Comment:
Are there plans to change the subdomain?

Reply:
fixed 9c4c335